app/models/middleware_datasource.rb

```diff
@@ -3,4 +3,10 @@ class MiddlewareDatasource < ApplicationRecord
   belongs_to :middleware_server, :foreign_key => "server_id"
   acts_as_miq_taggable
   serialize :properties
+
+  include LiveMetricsMixin
+
+  def metrics_capture
+    @metrics_capture ||= ManageIQ::Providers::Hawkular::MiddlewareManager::LiveMetricsCapture.new(self)
```

Comment:
Yes, it is.
An instance of a model that supports LiveMetricsMixin should support its metrics_capture implementation.
A future enhancement could be to study how it looks like if the metrics_capture can be shared from the provider itself (if it makes sense).
But that would be good for a following/future PR.